library/NetworkControl.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import logging
+import ipaddress
+
+from library.common import *
+from urllib.parse import quote
+
+
+class NetworkControl(object):
+    def __init__(self, requester, machine_id, interface_name='eth1'):
+        self.log = logging.getLogger(__name__)
+        self.requester = requester
+        self.machine_id = machine_id
+        self.interface_name = interface_name
+        self.machine_ip = ''
+        try:
+            self.interface = self.get_interface()
+        except Exception as err:
+            raise err
+
+    def get_interface(self):
+        url = "/api/v1/machine/{}/interface".format(self.machine_id)
+        try:
+            interfaces = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        for i in interfaces:
```

Comment:
this loop is obscure. What does it look for and why is an internal index into an array any valid outside?

Reply:
As a machine can have multiple interfaces, it searches for one called by default 'eth1' and returns the structure out of the array of interfaces. No index involved, I'll change the name to be more explicit.

Reply:
A comment would also be useful :)